addons/gut/gut.gd

```diff
@@ -643,18 +643,28 @@ func _pass(text=''):
 func _fail(text=''):
 	_gui.add_failing()
 	if(_current_test != null):
-		var line_text = ''
-		# Inner classes don't get the line number set so don't print it
-		# since -1 isn't helpful
-		if(_current_test.line_number != -1):
-			line_text = '  at line ' + str(_current_test.line_number)
-			p(line_text, LOG_LEVEL_FAIL_ONLY)
-			# format for summary
-			line_text =  "\n    " + line_text
+		var line_text = '  at line '
+		var line_number = _extractLineNumber( _current_test)
+		line_text += str(line_number)
+		p(line_text, LOG_LEVEL_FAIL_ONLY)
+		# format for summary
+		line_text =  "\n    " + line_text
 
 		_new_summary.add_fail(_current_test.name, text + line_text)
 		_current_test.passed = false
 
+# Extracts the line number from curren stacktrace by matching the test case name
+func _extractLineNumber(current_test) -> int:
+	var stackTrace:Array = get_stack()
+	if(stackTrace!=null):
+		for index in stackTrace.size():
+			var line:Dictionary = stackTrace[index]
+			var function = line.get("function")
+			if function == current_test.name:
+				return line.get("line")
+	# fallback to original line
+	return current_test.line_number
```

Comment:
I'm not a fan of early `return`s in general.  At first glance it looks like this will always return `current_test.line_number` and that kind of thing can make debugging difficult down the road.  Change this method to do something like:
```
var to_return = current_test.line_number
...
                              to_return = line.get("line")
return to_return
```

Reply:
done